src/backend/database/migrations/tenant/2025_01_18_100835_drop_user_table.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up() {
+        // Check if the users table exists in the tenant connection
+        if (Schema::connection('tenant')->hasTable('users')) {
+            // Fetch records from tenant users table
+            $users = DB::connection('tenant')->table('users')->get();
+
+            // Insert users into main schema if the main users table exists
+            if (Schema::connection('micro_power_manager')->hasTable('users')) {
+                foreach ($users as $user) {
+                    DB::connection('micro_power_manager')->table('users')->insert([
+                        'id' => $user->id,
+                        'name' => $user->name,
+                        'company_id' => $user->company_id,
+                        'email' => $user->email,
+                        'password' => $user->password,
+                        'created_at' => $user->created_at,
+                        'updated_at' => $user->updated_at,
+                    ]);
+                }
```

Comment:
Could you add an `else` block here that makes this section fail if the core schema doesn't have the `users` table? Basically, we want to prevent the `tenant.users` table to be dropped if `micro_power_manager.users` doesn't exist.
This could happen, if for some reason the Tenant migration is run, before the root migration is run.